File: admin-server/src/main/kotlin/com/systec/umeet/admin/service/SysUserGroupService.java
package com.systec.umeet.admin.service;

import com.systec.umeet.admin.mapper.SysUserGroupMapper;
import com.systec.umeet.admin.mapper.SysUserGroupPkMapper;
import com.systec.umeet.admin.mapper.SysUserGroupRolePkMapper;
import com.systec.umeet.admin.model.SysUserGroup;
import com.systec.umeet.admin.model.SysUserGroupPk;
import com.systec.umeet.admin.model.SysUserGroupRolePk;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * @description:
 * @author: super.wu
 * @date: Created in 2018/5/9 0009
 * @modified By:
 * @version: 1.0
 **/
@Service
@Transactional(rollbackFor = Exception.class)
public class SysUserGroupService extends BaseService<SysUserGroupMapper, SysUserGroup> {

    @Autowired
    private SysUserGroupPkMapper sysUserGroupPkMapper;

    @Autowired
    private SysUserGroupRolePkMapper sysUserGroupRolePkMapper;

    @Override
    public void deleteById(Object id) {
        int groupId = Integer.valueOf(id.toString());
        // 删除用户组和用户关联
        SysUserGroupPk userGroupPk = new SysUserGroupPk();
        userGroupPk.setGroupId(groupId);
        sysUserGroupPkMapper.delete(userGroupPk);
        // 删除用户组和角色关联
        SysUserGroupRolePk userGroupRolePk = new SysUserGroupRolePk();
        userGroupRolePk.setGroupId(Integer.valueOf(id.toString()));
        sysUserGroupRolePkMapper.delete(userGroupRolePk);
        // 删除用户组
        super.deleteById(id);
    }

}

File: umeet-admin-server/src/main/kotlin/com/systec/umeet/service/SysUserGroupService.kt
package com.systec.umeet.service
import com.github.pagehelper.PageInfo
import com.systec.umeet.model.SysUser
import com.systec.umeet.base.BaseServiceImpl
import com.systec.umeet.contants.ResponsePage
import com.systec.umeet.dao.SysUserDao
import com.systec.umeet.dao.SysUserGroupDao
import com.systec.umeet.dao.SysUserGroupPkDao
import com.systec.umeet.dao.SysUserGroupRolePkDao
import com.systec.umeet.model.SysUserGroup
import com.systec.umeet.model.SysUserGroupPk
import com.systec.umeet.model.SysUserGroupRolePk
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import tk.mybatis.mapper.common.Mapper

@Service
class SysUserGroupService : BaseServiceImpl<SysUserGroup> (){

    @Autowired
    var sysUserGroupDao : SysUserGroupDao? = null

    @Autowired
    var sysUserGroupPkDao : SysUserGroupPkDao ?= null

    @Autowired
    var sysUserGroupRoleDao : SysUserGroupRolePkDao ?= null

    override fun getMapper(): Mapper<SysUserGroup> {
        return sysUserGroupDao!!
    }

    /**
     * 根据条件分页查询用户
     */
    fun getPage(record: SysUserGroup): ResponsePage<SysUserGroup> {
        return super.selectPage(record.page , record.size ,record)
    }

    fun deleteById(id: Any) {
        val groupId = Integer.valueOf(id.toString())
        // 删除用户组和用户关联
        val userGroupPk = SysUserGroupPk()
        userGroupPk.groupId = groupId
        sysUserGroupPkDao!!.delete(userGroupPk)
        // 删除用户组和角色关联
        val userGroupRolePk = SysUserGroupRolePk()
        userGroupRolePk.groupId = Integer.valueOf(id.toString())
        sysUserGroupRoleDao!!.delete(userGroupRolePk)
        // 删除用户组
        super.deleteByPrimaryKey(id)
    }

}
